lesson_22/sources/string_utils.cpp

```diff
@@ -0,0 +1,78 @@
+#include <string>
+#include <sstream>
+#include <set>
+#include <unordered_map>
+#include <stack>
+
+#include "string_utils.hpp"
+
+
+unsigned int unique_words_count(const std::string& line)
+{
+    std::istringstream iss{ line };
+    std::set<std::string> unique_words;
+
+    std::string word;
+    while (iss >> word)
+    {
+        unique_words.insert(word);
+    }
+
+    return static_cast<unsigned int>(unique_words.size());
+}
+
+std::string most_occurred_word(const std::string& line)
+{
+    std::unordered_map<std::string, unsigned> word_count;
+
+    std::istringstream iss{ line };
+    std::string word;
+
+    std::string most_common_word;
+    unsigned max_count{ 0 };
+
+    while (iss >> word)
+    {
+        const unsigned current_count = ++word_count[word];
+
+        if(current_count > max_count)
+        {
+            max_count = current_count;
+            most_common_word = word;
+        }
```

Comment:
if you fill the map in one loop and then find maximum in other loop it takes less checks